Plugins/Sycl/include/Acts/Plugins/Sycl/Seeding/Seedfinder.ipp

```diff
@@ -0,0 +1,123 @@
+  // convert pT to p once theta angle is known
+  m_config.highland = 13.6 * std::sqrt(m_config.radLengthPerSeed) *
+                      (1 + 0.038 * std::log(m_config.radLengthPerSeed));
+  float maxScatteringAngle = m_config.highland / m_config.minPt;
+  m_config.maxScatteringAngle2 = maxScatteringAngle * maxScatteringAngle;
+  // helix radius in homogeneous magnetic field. Units are Kilotesla, MeV and
+  // millimeter
+  // TODO: change using ACTS units
+  m_config.pTPerHelixRadius = 300. * m_config.bFieldInZ;
+  m_config.minHelixDiameter2 =
+      std::pow(m_config.minPt * 2 / m_config.pTPerHelixRadius, 2);
+  m_config.pT2perRadius =
+      std::pow(m_config.highland / m_config.pTPerHelixRadius, 2);
+
+  m_queue = createQueue(device_name_substring);
+}
+
+template <typename external_spacepoint_t>
+template <typename sp_range_t>
+std::vector<Acts::Seed<external_spacepoint_t>>
+Seedfinder<external_spacepoint_t>::createSeedsForGroup(
+    sp_range_t bottomSPs, sp_range_t middleSPs, sp_range_t topSPs) const {
+  std::vector<Seed<external_spacepoint_t>> outputVec;
+
+  std::vector<deviceSpacePoint> deviceBottomSPs;
+  std::vector<deviceSpacePoint> deviceMiddleSPs;
+  std::vector<deviceSpacePoint> deviceTopSPs;
+
+  std::vector<const Acts::InternalSpacePoint<external_spacepoint_t>*>
+      bottomSPvec;
+  std::vector<const Acts::InternalSpacePoint<external_spacepoint_t>*>
+      middleSPvec;
+  std::vector<const Acts::InternalSpacePoint<external_spacepoint_t>*> topSPvec;
+
+  for (auto SP : bottomSPs) {
+    bottomSPvec.push_back(SP);
+    deviceBottomSPs.insert(
+        deviceBottomSPs.end(),
+        deviceSpacePoint{SP->x(), SP->y(), SP->z(), SP->radius(),
+                         SP->varianceR(), SP->varianceZ()});
+  }
+
+  for (auto SP : middleSPs) {
+    middleSPvec.push_back(SP);
+    deviceMiddleSPs.insert(
+        deviceMiddleSPs.end(),
+        deviceSpacePoint{SP->x(), SP->y(), SP->z(), SP->radius(),
+                         SP->varianceR(), SP->varianceZ()});
+  }
+
+  for (auto SP : topSPs) {
+    topSPvec.push_back(SP);
+    deviceTopSPs.insert(
+        deviceTopSPs.end(),
+        deviceSpacePoint{SP->x(), SP->y(), SP->z(), SP->radius(),
+                         SP->varianceR(), SP->varianceZ()});
+  }
+
+  const deviceSeedfinderConfig deviceConfigData = {
+      m_config.deltaRMin,
+      m_config.deltaRMax,
+      m_config.cotThetaMax,
+      m_config.collisionRegionMin,
+      m_config.collisionRegionMax,
+      m_config.maxScatteringAngle2,
+      m_config.sigmaScattering,
+      m_config.minHelixDiameter2,
+      m_config.pT2perRadius,
+      m_config.seedFilter->getSeedFilterConfig().deltaInvHelixDiameter,
+      m_config.seedFilter->getSeedFilterConfig().impactWeightFactor,
+      m_config.seedFilter->getSeedFilterConfig().deltaRMin,
+      m_config.seedFilter->getSeedFilterConfig().compatSeedWeight,
+      m_config.impactMax,
+      m_config.seedFilter->getSeedFilterConfig().compatSeedLimit,
+  };
```

Comment:
Why don't you make this object a member of `Acts::Sycl::SeedFinder`? There is no need to create it from scratch on every call.

Reply:
I did not want to change the interface of the Seedfinder class too much.

Reply:
:confused: You don't have to. You don't even need to change the list of arguments received by the constructor.

Just created a variable called `Acts::Sycl::SeedFinder::m_deviceConfigData`, and set it up in the constructor of the class. With this exact payload.

Reply:
Ok, I'll change this in the next commit.